app/hotels/src/filter/hotelFacilities/HotelFacilitiesPopup.js

```diff
@@ -0,0 +1,113 @@
+// @flow
+
+import * as React from 'react';
+import { Text, StyleSheet, View } from 'react-native';
+import {
+  ButtonPopup,
+  Color,
+  Checkbox,
+  Icon,
+} from '@kiwicom/react-native-app-common';
+
+type Props = {|
+  facilities: string[],
+  onClose: () => void,
+  onSave: (string[]) => void,
+  isVisible: boolean,
+|};
+
+type State = {|
+  facilities: string[],
+|};
+
+const facilitiesList = {
+  airportShuttle: { text: 'Airport shuttle', icon: 'airport-shuttle' },
+  familyRooms: { text: 'Family rooms', icon: 'child-friendly' },
+  facilitiesForDisabled: {
+    text: 'Facilities for Disabled Guests',
+    icon: 'accessible',
+  },
+  fitnessCenter: { text: 'Fitness Center', icon: 'fitness-center' },
+  parking: { text: 'Parking', icon: 'directions-car' },
+  freeParking: { text: 'Free Parking', icon: 'directions-car' },
+  valetParking: { text: 'Valet Parking', icon: 'directions-car' },
+  indoorPool: { text: 'Indoor Pool', icon: 'pool' },
+  petsAllowed: { text: 'Pets allowed', icon: 'pets' },
+  spa: { text: 'Spa', icon: 'spa' },
+  wifi: { text: 'Wi-Fi', icon: 'wifi' },
+};
+
+export default class HotelFacilitiesPopup extends React.Component<
+  Props,
+  State,
+> {
+  state = {
+    facilities: this.props.facilities,
+  };
+
+  componentWillReceiveProps = ({ facilities }: Props) =>
+    this.setState({ facilities });
+
+  handleCheckboxOnPress = (option: string) => () =>
+    this.setState(state => {
+      let facilities = [...state.facilities];
+      if (facilities.indexOf(option) >= 0) {
+        facilities = facilities.filter(facility => facility !== option);
+      } else {
+        facilities.push(option);
+      }
+      return { facilities };
+    });
+
+  onSave = () => this.props.onSave(this.state.facilities);
+
+  renderCheckboxes = (facilities: string[]) => {
+    const checkboxes = [];
+    Object.keys(facilitiesList).forEach((key, i) => {
+      const facility = facilitiesList[key];
+      checkboxes.push(
+        <Checkbox
+          key={i}
+          isChecked={facilities.indexOf(key) >= 0}
+          onPress={this.handleCheckboxOnPress(key)}
+          style={i < Object.keys(facilitiesList).length - 1 && styles.delimiter}
```

Comment:
This is not very nice, is it? It's very hard to read (and therefore understand) and moreover, you are iterating over `Object.keys` and calling the same `Object.keys` in every iteration. Can you please next time try more to write codes even for your colleagues (not only me)? I honestly don't like to do CR of your PRs because it's usually very very complicated (even though it usually works which is great)...

Also the whole `this.renderCheckboxes` is a little bit odd. It could've been much nicer component instead of function.